src/renderer/entities/signatory/lib/utils.ts

```diff
@@ -0,0 +1,22 @@
+import keyBy from 'lodash/keyBy';
+
+import { Account, AccountId, Wallet } from '@renderer/shared/core';
+import { walletUtils } from '@renderer/entities/wallet';
+
+export const singnatoryUtils = {
+  getSignatoryWallet,
+};
+
+function getSignatoryWallet(accoutId: AccountId, accounts: Account[], wallets: Wallet[]): Wallet | undefined {
```

Comment:
More meaningful parameters order, 2 fast returns
please check `return walletsMap[signatoryAccount?.walletId];`, I guess it should work 🤔
```javascript
function getSignatoryWallet(wallets: Wallet[], accounts: Account[], accoutId: AccountId): Wallet | undefined {
  const walletsMap = keyBy(wallets, 'id');

  const signatoryAccount = accounts.find((account) => {
    const accountIdMatch = accoutId === account.accountId;
    const wallet = walletsMap[account.walletId];
    
    if (!accountIdMatch || !wallet) return;

    return walletUtils.isValidSignatory(wallet) || walletUtils.isMultiShard(wallet);
  });

  return walletsMap[signatoryAccount?.walletId];
}
```

Reply:
`return walletsMap[signatoryAccount?.walletId]`
typescript error - cant use undefimed as index type